src/main/java/seedu/address/model/person/ModuleRoleMap.java

```diff
@@ -0,0 +1,232 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * Represents the mapping between module and role type in the roles
+ * taken by a Person in NUS.
+ */
+public class ModuleRoleMap {
+    /* Constraints message for role type. */
+    public static final String MESSAGE_CONSTRAINTS =
+            "Role consists of two parts, module code and role type.\n"
+            + "Role type can be blank, and or any of the following values(case insensitive) for each role type:\n"
+            + "1. Student: student or leave blank\n"
+            + "2. Tutor: tutor or ta\n"
+            + "3. Professor: professor or prof";
+
+    /*
+     * Input keyword must match one of the keyword provided in the regex.
+     * Note that the pattern is only matching lower case, so the other methods
+     * need to convert user input to lower case first before validate the input
+     */
+    public static final String VALIDATION_REGEX = "^$|student|tutor|ta|professor|prof";
+
+    private final HashMap<ModuleCode, RoleType> roles;
+
+    /**
+     * Default constructor for a {@code ModuleRoleMap}.
+     * This is for creating roles for a Person based on the module code and role types
+     * extracted from an add command.
+     *
+     * @param moduleCodes Array of modules codes
+     * @param roleTypes Array of role type String
+     */
+    public ModuleRoleMap(ModuleCode[] moduleCodes, RoleType[] roleTypes) {
+        requireNonNull(moduleCodes);
+        requireNonNull(roleTypes);
+
+        if (moduleCodes.length != roleTypes.length) {
+            throw new IllegalArgumentException(
+                    "Module code array's length is not consistent with that of role type array.");
+        }
+
+        checkArgument(isValidModuleCodeArray(moduleCodes), ModuleCode.MESSAGE_CONSTRAINTS);
```

Comment:
Is it possible to write another method to check for null object in an array(to avoid duplicate codes)?
Also, the MESSAGE_CONSTRAINTS I think should be changed to some other error message like "Null object not allowed" since this essentially checks for that. (though Im not sure under what circumstances will there be null object)

Reply:
Hi, I agree with the suggestion on avoiding code duplication, but regarding `MESSAGE_CONSTRAINTS` I think they are actually displayed to the users, and telling them arrays contains null objects are not very useful and instead we should tell them how to input data properly to avoid null objects in the generated arrays. Would you think it would be better to leave the MESSAGE_CONSTRAINTS as it is?

Reply:
Make sense! Since add module not implemented yet, we are not sure what kind of input can cause this to happen